src/vehicle.cpp

```diff
@@ -3381,13 +3387,23 @@ void vehicle::consume_fuel( double load = 1.0 )
 
         //get exact amount of fuel needed
         double amnt_precise = double(amnt_fuel_use) / ft.coeff;
-
         amnt_precise *= load * (1.0 + st * st * 100);
+        int remainder = int( amnt_precise * 1000 ) % 1000;
         int amnt = int(amnt_precise);
-        // consumption remainder results in chance at additional fuel consumption
-        if( x_in_y(int(amnt_precise*1000) % 1000, 1000) ) {
-            amnt += 1;
+
+        if( remainder != 0 ) {
+            auto previous_remainder = fuel_remainder.find( ft.id );
+            if( previous_remainder != fuel_remainder.end() ) {
+                previous_remainder->second += remainder;
+		if( previous_remainder->second >= 1000 ) {
+                    previous_remainder->second -= 1000;
+                    amnt++;
```

Comment:
Isn't `amnt` suppose to be amount of fuel NEEDED (for this tick) rather than available? From what I can read here, remainder getting to 1000 will cause a tick with +1 fuel used.